### DISCUSSION TITLE
decouple channel_layer from channels

### DISCUSSION THREAD

**Discussion by lsaavedr** [django/channels#1571] [Ideas] [answered]
Hi @carltongibson, thanks for your beautiful code,

I'm working with channels and is one of the wonderful things in django, at the same time I'm working with an udp-socket server (to some particular IoT and cannot be this changed), I'm searching to some fast (like UDP) to connect this udp-socket server to my django backend and I'm trying with channel-layer, the good part is that it works like a charm but the bad things is that to use channel-layer with channel-redis I'm need to install channels->(daphne & django) 😔, because of this I think that is a good thing decouple channel_layer to channels (also daphne to channels),

Regards,
Luis

**Comment by carltongibson** [comment]
Hi @lsaavedr the channel layer logic is all in one file https://github.com/django/channels/blob/master/channels/layers.py — you'd be welcome to extract what you need from that into your own project.

**Comment by lsaavedr** [comment]
Yes, extract that is straightforward, but I'm talking about the dependence structure, to use any of this pieces of software we need to install django and daphne (for example)

**Comment by smorokin** [comment, marked as answer]
I think [nejma](https://github.com/taoufik07/nejma) is a standalone `channel_layer` substitue.

**Reply by carltongibson** [reply]
Super. That might be the way to go. 

From my perspective here there is simply not the bandwidth for the work to maintain separate packages. 
The channels footprint is not so big that this is a concern we can worry about .

**Comment by lsaavedr** [comment]
nejma seems to be unmaintained :(

**Reply by carltongibson** [reply]
@lsaavedr I think you’re worrying about nothing here. Just install Channels and use the bits you want. Same with Django: you don’t have to use the bits you’re not interested in. The files take up a few KB on the hard drive, no biggie. (Certainly not worth the effort of maintaining multiple packages — much better to put that time into the one.)
